client.go

```diff
@@ -0,0 +1,103 @@
+package gotruevault
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+)
+
+const (
+	contentTypeApplicationJSON = "application/json"
+)
+
+var (
+	// ErrUnauthorized ...
+	ErrUnauthorized = errors.New("error: authorization failed")
+
+	// ErrServerError ...
+	ErrServerError = errors.New("error: server error")
+
+	// ErrBadRequest ...
+	ErrBadRequest = errors.New("error: bad request")
+)
+
+// Client is the interface that wraps http calls to TrueVault
+//go:generate mockery --name Client
+type Client interface {
+	SearchDocument(ctx context.Context, vaultID string, filter SearchFilter) (SearchDocumentResult, error)
+}
+
+type trueVaultClient struct {
+	httpClient *http.Client
+	urlBuilder URLBuilder
+	apiKey     string
```

Comment:
we also need to support accessTokens but we can add support for that in another PR

Reply:
Added support for access token in this commit https://github.com/FirstVisit/go-truevault/pull/1/commits/e8d658aa8979fd0973cf70de1e9b34f4adf2852c